### DISCUSSION TITLE
記事本文の右の柱の自動生成された目次から、一部の見出しが抜けることがある

### DISCUSSION THREAD

**Discussion by kkkkan** [increments/qiita-discussions#178] [要望・アイデア・不具合]
markdownのパースはちゃんとされていて、見出しは見出しとして表示されているのに画面右の柱の自動生成される目次からは一部の見出しが抜けることがある気がします。

例えば以下リンク↓

https://qiita.com/kkkkan/items/1025a5abf3d0fd4b6d7e
の2022年03月15日 22時33分 (JST)の編集後の状態など。

↓本文部分のスクリーンショット。見出しは1~4まであります。
![image](https://user-images.githubusercontent.com/22609306/158389109-d9eb7f2e-fbfb-428f-9338-85574310f0f8.png)

↓目次部分のスクリーンショット。見出しは1と4しか含まれていません。
![image](https://user-images.githubusercontent.com/22609306/158388050-ad977a0f-9317-4a22-b6b4-a149c84e241d.png)

**Comment by kyntk** [comment]
@kkkkan ご報告ありがとうございます！

こちらは、新しいパーサーによって発生した不具合のようです :bow: 
今回の記事については、
~~~
- リスト3
```gradle
~~~
の間に空行を入れることで治ると思われます :pray: 

不具合については原因調査と対応をいたします:bow:

**Comment by kkkkan** [comment]
@kyntk  
調査ありがとうございます。了解です！
お手数ですがよろしくお願いいたします。

修正版がリリースがされましたらお手数をおかけしますが、ここでの返信の形でいいのでご連絡いただけると嬉しいです！(テスト投稿の記事を削除したいので)
ご面倒おかけしますがよろしくお願いします🙇

**Comment by kkkkan** [comment]
他の記事でも再現したので、念のため報告いたします。

以下の記事の
https://qiita.com/kkkkan/items/2b32f011236d64acdcb9

# キャンセルの伝播の仕組み
## 前提
## 誰が伝播を担うのか？
## 伝播の方向
## `CoroutineScope#cancel` の実装
## `CoroutineScope`の中で例外が発生した時
### `Job`要素がない`CoroutineContext`から作成された`CoroutineScope`は？
# `CorouitneScope#launch`でのキャンセルの伝播の様子を内部実装から追ってみる
## 初めに、`launch`内で例外が発生した時の実際のコード出力を見てみる

が目次から抜けているようです。(2022年03月27日 11時13分 (JST)編集時点)

**Reply by kyntk** [reply]
@kkkkan 追加のご報告ありがとうございます。

不具合の確認してみたのですが、ToCではCode blockの前後に空行が無いと適切にパースされないことが分かりました。
ただ、不具合修正には時間がかかり、すぐに解決！とできなそうで、お時間をいただきたいです 🙇🏽 

それまではお手数ですが、ToCがおかしいなと思ったらCode blockの前後に空行をいれていただけますと幸いです。

**Reply by kkkkan** [reply]
@kyntk 

>ToCではCode blockの前後に空行が無いと適切にパースされないことが分かりました。
原因は分かったのとのことよかったです。

> 不具合修正には時間がかかり、すぐに解決！とできなそうで、お時間をいただきたいです 🙇🏽
承知いたしました。
お手数をおかけしますが、よろしくお願いいたします。🙇🏽

Code blockの前後に空行をいれるようにいたします！

**Comment by kyntk** [comment]
こちらでも同様の報告をいただいた
https://github.com/increments/qiita-discussions/discussions/202

**Comment by scivola** [comment]
（関連して）
`*` などによる箇条書きの直後に（空行を入れずに）見出しが続くとき，当該の見出しが目次に反映されないようですね。

**Comment by scivola** [comment]
コードブロックの前後に空行が入っているのに見出しが抜け，逆にコードブロック中の `#` で始まる行を見出しとして拾ってしまっている例を見つけました。
https://qiita.com/onumakunz/items/149af0b6d609ceea200d
（2022/05/27 20:18 JST 閲覧）

**Reply by kyntk** [reply]
@scivola ご報告ありがとうございます！
ただいま、この不具合解消に着手したところです！

お待たせしてしまい、申し訳ございません！ 🙇

**Comment by kyntk** [comment]
@kkkkan @scivola 
目次の不具合修正が完了いたしました！
お待たせして申し訳ございませんでした 🙇 

こちらの記事の目次も正しく出るようになりました！
https://qiita.com/kkkkan/items/1025a5abf3d0fd4b6d7e

**Comment by kkkkan** [comment]
@kyntk 
確認遅れました。すみません。

修正ありがとうございます！
正しく表示されるようになっていることも確認しました！
ご対応いただきありがとうございました。🙇
